integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,278 @@
+  private final long mStatCacheTimeout;
+  private final boolean mUserGroupTranslationEnabled;
+
+  /**
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with only Alluxio cluster configuration.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(AlluxioConfiguration conf) {
+    Preconditions.checkNotNull(conf,
+        "AlluxioConfiguration should not be null for creating AlluxioFuseFileSystemOpts.");
+    List<String> libfuseOptions = conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)
+        ? conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
+        : ImmutableList.of();
+    Optional<String> authPolicyCustomGroup = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP)
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP))
+        : Optional.empty();
+    Optional<String> authPolicyCustomUser = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER)
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER))
+        : Optional.empty();
+    return new AlluxioFuseFileSystemOpts(
+        conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH),
+        conf.getString(PropertyKey.FUSE_FS_NAME),
+        conf.getClass(PropertyKey.FUSE_AUTH_POLICY_CLASS),
+        authPolicyCustomGroup,
+        authPolicyCustomUser,
+        conf.getInt(PropertyKey.FUSE_CACHED_PATHS_MAX),
+        (int) conf.getMs(PropertyKey.FUSE_UMOUNT_TIMEOUT),
+        conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED),
+        libfuseOptions,
+        conf.getBoolean(PropertyKey.USER_METADATA_CACHE_ENABLED),
+        conf.getString(PropertyKey.FUSE_MOUNT_POINT),
+        conf.getBoolean(PropertyKey.FUSE_SPECIAL_COMMAND_ENABLED),
+        conf.getMs(PropertyKey.FUSE_STAT_CACHE_REFRESH_INTERVAL),
+        conf.getBoolean(PropertyKey.FUSE_USER_GROUP_TRANSLATION_ENABLED)
+    );
+  }
+
+  /**SystemUserGroupAuthPolicy.java
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with
+   * Alluxio cluster configuration and command line input.
+   * Command line input has higher precedence if a property is set both in config and command.
+   *
+   * @param conf Alluxio cluster configuration
+   * @param fuseCliOpts Alluxio fuse command line input
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(
+      AlluxioConfiguration conf, AlluxioFuseCliOpts fuseCliOpts) {
+    Preconditions.checkNotNull(conf,
+        "AlluxioConfiguration should not be null for creating AlluxioFuseFileSystemOpts.");
+    Preconditions.checkNotNull(fuseCliOpts,
+        "AlluxioFuseCliOpts should not be null for creating AlluxioFuseFileSystemOpts.");
+    String alluxioPath = fuseCliOpts.getMountAlluxioPath().orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH));
+    Optional<String> authPolicyCustomGroup = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP)
```

Comment:
why not use Optional.OfNullable? can it assign Optional.empty() to empty string case?

Reply:
`conf.getString()` throws exception if the value is not set, instead of null or an empty string.